src/views/Player.vue

```diff
@@ -105,15 +115,32 @@ export default defineComponent({
       hasQuizEnded: false, // whether the quiz has ended - only valid for quizType = assessment
       sessionId: "", // id of the session created for a user-quiz combination
     });
+
+    const requiresAuth = computed(() => {
+      return props.userId != null && props.apiKey != null;
+    });
+
+    async function getAuth() {
+      if (requiresAuth.value) {
+        UserAPIService.checkAuthToken(props.apiKey).catch(() => {
+          router.replace({
+            name: "403",
+          });
+        });
+      }
+    }
+
+    getAuth();
+
     const isQuizAssessment = computed(
       () => state.metadata.quiz_type == "assessment"
     );
+
     const isSplashShown = computed(() => state.currentQuestionIndex == -1);
     const numQuestions = computed(() => state.questions.length);
     const isQuestionShown = computed(() => {
       return (
-        state.currentQuestionIndex >= 0 &&
-        state.currentQuestionIndex < numQuestions.value
+        state.currentQuestionIndex >= 0 && state.currentQuestionIndex < numQuestions.value
```

Comment:
please check your linter settings - we should all have the same linter settings or else these non-changes will keep showing up as changes and make reviewing a pain. Please check with me or Deepansh for our linter settings.